### ISSUE TITLE
DiscreteFourier2D type error

### ISSUE THREAD

**Issue by swkeemink** [losonczylab/sima#242]
The following error pops up on up-to-date python 2 installations when  using sima.motion.DiscreteFourier2D() to motion correct (other methods work fine):
```
/home/sander/Code/rochefort-lab/sima/sima/motion/motion.py in _observation_counts(raw_shape, displacements, untrimmed_shape)
    250             cnt[plane + d[0],
    251                 d[1]:(d[1] + raw_shape[1]),
--> 252                 d[2]:(d[2] + raw_shape[2])] += 1
    253     elif displacements.ndim == 3:
    254         if displacements.shape[-1] == 2:

TypeError: slice indices must be integers or None or have an __index__ method
```
It seems to come from d containing floats when using the discrete fourier function. It could be fixed by putting a line here to enforce d to contain only ints, but it seems to be a problem specific for the discrete fourier function.

On our fork I was able to fix it by making sure the output of the ``` DiscreteFourier2D``` class is of integer type: https://github.com/rochefort-lab/sima/commit/0c8faae21918156da2d5bc22e2a7d1678fc9cec0
I'm happy to make a pull request for it, but perhaps there is a more elegant way to fix this, or I've made a mistake with my installation, or it is actually desirable to have a float output, etc..

The error was reproduced across up-to-date Anaconda, WinPython, and general Python installs under Ubuntu, OS X and Windows. Funnily enough, on some of our older WinPython installations we do not run into this issue, so it might be related to some new numpy/scipy behaviour, which I've not been able to identify yet.

Apologies if this relates to an existing issue that I overlooked.

**Comment by jzaremba** [comment]
Numpy used to allow you to index arrays with floats, but that changed at some point, which is why this is only causing problems with newer installs.

The DFT registration can also be used for sub-pixel registration (`upsample_factor` argument), so I think the best option is to round and then convert to int right before you do the counting. 

https://github.com/losonczylab/sima/blob/24694b94c22f95b16659236ce311d698e4251e98/sima/motion/motion.py#L245-L252

Between lines 249 and 250, something like this should work:

``d = np.round(d).astype('int')``

I haven't used DFT registration myself, but if this works and you want to setup a pull-request, that would be great.

**Comment by james-priestley** [comment]
See my pull request, but I fixed this instead in MotionEstimationStrategy.correct, which should also avoid truncating non-integer shifts instead of rounding when applying the displacements due to the handling in _MotionCorrectedSequence.

**Comment by jzaremba** [comment]
Fixed by #250
